File: swift_calculator/calculator/calculator/CalculatorConstants.swift
//
//  CalculatorConstants.swift
//  calculator
//
//  Created by JOSE ZARABANDA on 7/25/25.
//

import SwiftUI

struct CalculatorConstants {
    // UI Constants
    static let buttonSize: CGFloat = 80
    static let buttonSpacing: CGFloat = 12
    static let cornerRadius: CGFloat = 40
    static let displayFontSize: CGFloat = 64
    static let buttonFontSize: CGFloat = 28
    
    // Maximum display length
    static let maxDisplayLength = 9
    
    // Number formatting
    static let maxFractionDigits = 8
    static let scientificNotationThreshold = 1e15
    static let smallNumberThreshold = 1e-6
}

extension Color {
    static let calculatorBackground = Color.black
    static let calculatorText = Color.white
}

struct CalculatorColors {
    static let background = Color.black
    static let text = Color.white
    static let numberButton = Color(red: 0.2, green: 0.2, blue: 0.2)
    static let operationButton = Color.orange
    static let functionButton = Color(red: 0.6, green: 0.6, blue: 0.6)
}

File: fluter_calculator/lib/calculator_constants.dart
import 'package:flutter/material.dart';

class CalculatorConstants {
  // UI Constants
  static const double buttonSize = 80.0;
  static const double buttonSpacing = 12.0;
  static const double cornerRadius = 40.0;
  static const double displayFontSize = 64.0;
  static const double buttonFontSize = 28.0;
  
  // Maximum display length
  static const int maxDisplayLength = 9;
  
  // Number formatting
  static const int maxFractionDigits = 8;
  static const double scientificNotationThreshold = 1e15;
  static const double smallNumberThreshold = 1e-6;
}

class CalculatorColors {
  static const Color background = Colors.black;
  static const Color text = Colors.white;
  static const Color numberButton = Color(0xFF333333); // Color(red: 0.2, green: 0.2, blue: 0.2)
  static const Color operationButton = Colors.orange;
  static const Color functionButton = Color(0xFF999999); // Color(red: 0.6, green: 0.6, blue: 0.6)
}
